m4/cf3_platforms.m4

```diff
@@ -39,3 +39,4 @@ AM_CONDITIONAL([HPUX], [test -n "`(echo ${target_os} | egrep 'hpux|hp-ux')`"])
 AM_CONDITIONAL([FREEBSD], [test -n "`(echo ${target_os} | grep freebsd)`"])
 AM_CONDITIONAL([NETBSD], [test -n "`(echo ${target_os} | grep netbsd)`"])
 AM_CONDITIONAL([XNU], [test -n "`(echo ${target_os} | grep darwin)`"])
+AM_CONDITIONAL([SHUNIT], [test -n "`which shunit2`"])
```

Comment:
Got this working cross-platform without toooo much hassle, but I concede it got a bit uglier. `egrep` is well supported (better than `grep -E`, which requires PATH fiddling in Solaris). `sed -E` is not so I rewrote those statements to not need it. I checked docs for AIX and HP-UX, and tested on Solaris, and I'm pretty confident this is portable. Need help testing though.

If there are strong opinions that the previous form was nicer, I think another elegant option would be to limit the platforms we run the tests on by testing sed / grep's capabilities in the autoconf stuff.